### DISCUSSION TITLE
Feature: Check if stream is online

### DISCUSSION THREAD

**Discussion by ibex-are-goats** [streamlink/streamlink#3715] [Q&A] [answered]
Just a simple binary check and print: 0 if stream is offline, 1 if online.

**Comment by bastimeyer** [comment, marked as answer]
streamlink_cli already returns 0 on success and 1 on error

```bash
#!/usr/bin/env bash
streamlink stream-that-does-not-exist >/dev/null
echo $?
1
```

```bash
#!/usr/bin/env bash
streamlink stream-that-does-exist >/dev/null
echo $?
0
```

```bash
#!/usr/bin/env bash
if streamlink stream-that-does-exist >/dev/null; then
  echo live
else
  echo not live
fi
```

```bash
#!/usr/bin/env bash
streamlink stream-that-does-exist >/dev/null && echo live || echo not live
```

and via streamlink's API, you can just check the length of the returned streams length

```py
#!/usr/bin/env python
>>> import streamlink
>>> len(streamlink.streams("stream-that-does-not-exist")) > 0
False
>>> len(streamlink.streams("stream-that-does-exist")) > 0
True
```

**Comment by bastimeyer** [comment]
But depending on the website/service and stream you're checking, instead of using Streamlink for this, it's almost always better to use the service's public web API if there is one available.

**Comment by ibex-are-goats** [comment]
Oh, that's a great feature! Thank you! You could put that right in the documentation.

About using the Twitch API, I'm just a novice scripter, so I need to call an app that does the smart stuff for me ;)

**Comment by bastimeyer** [comment]
> Twitch

Don't use Streamlink for this. It's very inefficient.

Use their REST API (kraken v5):

```bash
#!/usr/bin/env bash
set -e
CHANNEL=$1
[[ -n "${CHANNEL}" ]] || exit 1
HEADERS=(
  -H "Accept: application/vnd.twitchtv.v5+json"
  -H "Client-ID: YOUR_API_CLIENT_ID"
)
USER_ID=$(curl -sSL "${HEADERS[@]}" \
  "https://api.twitch.tv/kraken/users?login=${CHANNEL}" \
  | jq -r '.users[0]._id'
)
curl -sSL "${HEADERS[@]}" \
  "https://api.twitch.tv/kraken/streams/${USER_ID}" \
  | jq -e '.stream._id' >/dev/null
```

```bash
./twitch.sh esl_dota2; echo $?
0
./twitch.sh a-channel-that-does-not-exist; echo $?
1
```

**Reply by ibex-are-goats** [reply]
I've been using your script for the last couple months and it works great!

Now I'm trying to convert it to a python function, and I'm over my head conceptually, trying to use python request and managing the json. If it's doable for you, can you lay down the bones of it for me?

I actually just checked the TwitchAPI project first for this, and strangely enough, even they don't have a simple online check function of their own yet. https://pytwitchapi.readthedocs.io/en/latest/modules/twitchAPI.twitch.html

**Reply by bastimeyer** [reply]
Your question is out of scope of this project. Check the official Twitch API documentation.
https://dev.twitch.tv/docs/
